x/concentrated-liquidity/incentives_test.go

```diff
@@ -3509,3 +3511,241 @@ func (s *KeeperTestSuite) TestPrepareBalancerPoolAsFullRange() {
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets: []balancer.PoolAsset{
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("foo", sdk.NewInt(200))},
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("bar", sdk.NewInt(250))},
+			},
+			uptimeGrowth: uptimeHelper.emptyExpectedAccumValues,
+		},
+		"balancer spot price different than CL spot price (bar higher)": {
+			// 100 existing shares and 200 shares added from balancer
+			// Note that only 200foo/200bar qualify, and the remaining 50foo is not counted
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets: []balancer.PoolAsset{
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("foo", sdk.NewInt(250))},
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("bar", sdk.NewInt(200))},
+			},
+			uptimeGrowth: uptimeHelper.emptyExpectedAccumValues,
+		},
+		"rounding check: large and imbalanced CL amounts": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(2<<60)), sdk.NewCoin("bar", sdk.NewInt(2<<61))),
+			balancerPoolAssets:  defaultBalancerAssets,
+			uptimeGrowth:        uptimeHelper.hundredTokensMultiDenom,
+		},
+		"rounding check: large and imbalanced balancer amounts": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets: []balancer.PoolAsset{
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("foo", sdk.NewInt(2<<61))},
+				{Weight: sdk.NewInt(1), Token: sdk.NewCoin("bar", sdk.NewInt(2<<60))},
+			},
+			uptimeGrowth: uptimeHelper.hundredTokensMultiDenom,
+		},
+
+		// Error catching
+
+		"CL pool does not exist": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets:  defaultBalancerAssets,
+			uptimeGrowth:        uptimeHelper.hundredTokensMultiDenom,
+
+			concentratedPoolDoesNotExist: true,
+			expectedError:                types.PoolNotFoundError{PoolId: invalidPoolId + 1},
+		},
+		"Balancer pool does not exist": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets:  defaultBalancerAssets,
+			uptimeGrowth:        uptimeHelper.hundredTokensMultiDenom,
+
+			balancerPoolDoesNotExist: true,
+			expectedError:            poolincentivestypes.NoGaugeAssociatedWithPoolError{PoolId: invalidPoolId, Duration: longestLockableDuration},
+		},
+		"Balancer shares not yet added to CL pool accums": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets:  defaultBalancerAssets,
+			uptimeGrowth:        uptimeHelper.hundredTokensMultiDenom,
+
+			balSharesNotAddedToAccums: true,
+			expectedError:             types.BalancerRecordNotFoundError{ClPoolId: 1, BalancerPoolId: 2, UptimeIndex: uint64(0)},
+		},
+	}
```

Comment:
Would be good to have a test where claimed rewards end up being greater than what's available in the incentives address

Reply:
I think having `osmoassert.ConditionalPanic` and an extra boolean `expectedPanic` in the test case struct is sufficient to cover this test case.

Another approach is to manually check the balancers in the function and return an error if the balance is insufficient

Reply:
Turns out the panic is actually caught at a lower level of abstraction so it was actually even simpler than that to add!